src/Bicep.Core/Extensions/JsonElementExtensions.cs

```diff
@@ -155,5 +155,14 @@ private static void Merge(JsonElement first, JsonElement second, Utf8JsonWriter
         }
 
         private static string[] GetPropertyNames(string path) => path.Split('.');
+
+        public static string[]? TryGetStringArray(this JsonElement element)
+        {
+            if (element.ValueKind != JsonValueKind.Array)
+            {
+                return null;
+            }
+            return element.EnumerateArray().Select(x => x.GetString()).WhereNotNull().ToArray();
```

Comment:
Do we need to be resilient to authoring issues (e.g. setting this to a non-string value?)

Reply:
I think here the function `x.GetString()` returns `null` for non string values, then those are filtered by the `WhereNotNull()` so my understanding is that when anything that is not a string is found then it is ignored. I may be missing something though, can you elaborate what is the concern?